cypress/support/step-definitions/menu-list-steps.js

```diff
@@ -0,0 +1,41 @@
+import {
+  menuTitle, menuListPreview, menuListSearchInput,
+  menuSecondOption, menuList,
+} from '../../locators/menu-list';
+
+Then('title on preview is {string}', (title) => {
+  menuTitle().first().should('have.text', title);
+});
+
+When('I click into title', () => {
+  menuTitle().first().click();
+});
+
+When('I click into menu item second element', () => {
+  menuSecondOption().click();
+});
+
+Then('MenuList component is expanded', () => {
+  menuListPreview().children().should('have.length', 4);
+});
+
+Then('MenuList component is not expanded', () => {
+  menuListPreview().children().should('have.length', 0);
+});
+
+Then('filter is disabled', () => {
+  menuListSearchInput().should('not.exist');
+});
+
+Then('filter is enabled', () => {
+  menuListSearchInput().should('exist');
+});
+
+When('I change search parameter to {string}', (parameter) => {
+  menuListSearchInput().clear().type(parameter);
+});
+
+Then('search result is {string}', (parameter) => {
+  menuList().parent().should('have.length', 2);
```

Comment:
why `menuList().parent()` should have length equals 2? no information about it in step name or scenario

Reply:
it is just additional assertion, which checks that there are the search input and the found element

Reply:
I suggest to add separate step for results count assertion which I put it as scenario comment, please see there and tell me what you think :)

Reply:
added